starter_project/autonomy/src/perception.cpp

```diff
@@ -54,15 +61,60 @@ namespace mrover {
         tags.clear(); // Clear old tags in output vector
 
         // TODO: implement me!
+        // cv::aruco::ArucoDetector detector = cv::aruco::ArucoDetector(mTagDictionary, mTagDetectorParams);
+
+        cv::aruco::detectMarkers(image, mTagDictionary, mTagCorners, mTagIds, mTagDetectorParams);
+
+        std::vector<float> closeness_metrics;
```

Comment:
I think these collections are redundant. You have everything you need in `mTagCorners` and `mTagIds` right. The following two for loops could just be one